.github/workflows/test.yml

```diff
@@ -0,0 +1,88 @@
+name: Test
+
+on:
+  pull_request:
+    branches: [main]
+    paths: '**.py'
+  push:
+    branches: [main]
+    paths: '**.py'
+
+env:
+  # The lower bound from the pyproject.toml file
+  OLDEST_PYMC_VERSION: "$(grep -E 'pymc *>' pyproject.toml | sed -n 's/.*>=\\([0-9]*\\.[0-9]*\\.[0-9]*\\).*/\\1/p')"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        config: [ {python-version: "3.10", oldest-pymc: false}, {python-version: "3.12", oldest-pymc: true}]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.config.python-version }}
+      - name: Install oldest version of PyMC
+        if: ${{ matrix.config.oldest-pymc }}
+        run: pip install pymc==${{ env.OLDEST_PYMC_VERSION }}
+      - name: Run tests
+        run: |
+          pip install -e .[test]
+          pytest --cov-report=xml --no-cov-on-fail --durations=50
+      - name: Check oldest version of PyMC
+        if: ${{ matrix.config.oldest-pymc }}
+        run: python -c "import pymc; assert pymc.__version__ == '${{ env.OLDEST_PYMC_VERSION }}'"
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # use token for more robust uploads
+          name: ${{ matrix.config.python-version }}
+          fail_ci_if_error: false
+
+  test_slow:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Run tests
+        run: |
+          pip install -e .[test]
+          pytest --only-slow --cov-report=xml --no-cov-on-fail --durations=50
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }} # use token for more robust uploads
+          name: "test_slow"
+          fail_ci_if_error: false
+
+  example_notebooks:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.12"
+    - name: Install dependencies
+      run: |
+        sudo apt-get install graphviz
+        pip install -e .[docs]
+        pip install -e .[test]
+    - name: Run notebooks
+      run: make run_notebooks
+
+
+  all:
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    name: All checks
+    needs: [ test, test_slow ]
+    steps:
+      - name: Confirm checks passed
+        if: ${{ (needs.test.result != 'success' || needs.test_slow.result != 'success') }}
+        run: exit 1
```

Comment:
To be honest I did not write this part of the CI. I agree it is not bringing much (I hope we are not missing something)

Reply:
Maybe we can keep and remove the if: ${{ always() }}

CC: @ricardoV94

Reply:
The default behavior is exiting if either test or test_slow fail. Right now we have that functionality turned off with `fail_ci_if_error: false` in ci, so we have to check at the end there. I also don't see why we would want to do this.

Reply:
Good point. Lets remove that to make sure they fail the workflow

Reply:
The reason for delay failing is all jobs are run instead of stopping when either one fails. 

In my experience this ultimately saves on CI runs because we have a list of all failing tests that can be checked locally (since nobody runs the whole suite locally by default)

Reply:
Makes sense. Do you have a recommendation on this change?

Reply:
> Makes sense. Do you have a recommendation on this change?

Nope, didn't check the other changes to know what's the context. Just wanted to shine light on this bit